### PR TITLE
Move the watcher cli application to be a go routine

### PR DESCRIPTION
**Pull request by iblancasa**
This PR moves the `watcher` CLI application to be a goroutine.

We were discussing this change to avoid the need to create another image as a product.

### CODE REVIEW

**Comment by iblancasa** [comment]
@JoaoBraveCoding I'm still testing but, if you have time, can you start providing feedback?

File: main.go [hunk 1]
```diff
@@ -171,16 +158,28 @@ func runController(ctx context.Context, kubeConfig *rest.Config) error {
...
 	}
 
 	err = mgr.AddAgent(mcoaAgentAddon)
 	if err != nil {
-		klog.Fatal(err)
+		logger.Error(err, "unable to add agent")
+		return err
+	}
+
+	disableReconciliation := os.Getenv("DISABLE_RECONCILIATION")
```
Comment (iblancasa) [inline main.go]: Maybe this can be a CLI option. I think this can be interesting to do for testing reasons.

Comment (JoaoBraveCoding) [reply main.go]: Did you run into a use-case where you needed this feature? If we keep it I would change it to `DISABLE_WATCHER_CONTROLLER`. I'm fine with moving it to CLI option if we want to keep it

File: main.go [hunk 2]
```diff
@@ -171,16 +157,29 @@ func runController(ctx context.Context, kubeConfig *rest.Config) error {
...
 	if err != nil {
-		klog.Errorf("failed to build agent %v", err)
+		logger.Error(err, "failed to build agent")
 		return err
 	}
 
 	err = mgr.AddAgent(mcoaAgentAddon)
 	if err != nil {
-		klog.Fatal(err)
+		logger.Error(err, "unable to add agent")
```
Comment (JoaoBraveCoding) [inline main.go]:
```suggestion
		logger.Error(err, "unable to add mcoa agent addon")
```

File: main.go [hunk 3]
```diff
@@ -171,16 +157,29 @@ func runController(ctx context.Context, kubeConfig *rest.Config) error {
...
+		return err
+	}
+
+	disableReconciliation := os.Getenv("DISABLE_RECONCILIATION")
+	if disableReconciliation == "" {
+		var wm *watcher.WatcherManager
+		wm, err = watcher.NewWatcherManager(logger, scheme)
+		if err != nil {
+			logger.Error(err, "unable to create the reconciliation manager")
```
Comment (JoaoBraveCoding) [inline main.go]:
```suggestion
			logger.Error(err, "unable to create watcher manager")
```

File: internal/controllers/watcher/controller.go [hunk 4]
```diff
@@ -29,6 +31,54 @@ var noReconcilePred = builder.WithPredicates(predicate.Funcs{
...
+	wm := WatcherManager{
+		mgr:    &mgr,
+		logger: l,
+	}
+
+	return &wm, nil
+}
+
+func (wm *WatcherManager) Start(ctx context.Context) {
+	wm.logger.Info("Starting reconciliation manager")
```
Comment (JoaoBraveCoding) [inline internal/controllers/watcher/controller.go]:
```suggestion
	wm.logger.Info("Starting watcher manager")
```

File: internal/controllers/watcher/controller.go [hunk 5]
```diff
@@ -29,6 +31,54 @@ var noReconcilePred = builder.WithPredicates(predicate.Funcs{
...
+	go func() {
+		err := (*wm.mgr).Start(ctx)
+		if err != nil {
+			wm.logger.Error(err, "there was a problem while running the reconciliation watcher")
```
Comment (JoaoBraveCoding) [inline internal/controllers/watcher/controller.go]:
```suggestion
			wm.logger.Error(err, "there was an error while running the reconciliation watcher")
```

**Review by JoaoBraveCoding — COMMENTED** [review]
minor nits

**Comment by iblancasa** [comment]
> Did you run into a use-case where you needed this feature? If we keep it I would change it to DISABLE_WATCHER_CONTROLLER. I'm fine with moving it to CLI option if we want to keep it****

For testing, for instance.

**Comment by iblancasa** [comment]
@JoaoBraveCoding added your changes. I think we are ready to merge.

**Pull request merged**